packages/ubus-lime-utils/tests/test_lime_utils_admin.lua

```diff
@@ -55,33 +63,74 @@ describe('ubus-lime-utils-admin tests #ubuslimeutilsadmin', function()
         local response  = rpcd_call(ubus_lime_utils, {'call', 'firmware_verify'},
                                     '{"fw_path": "/foo"}')
         assert.is.equal("ok", response.status)
-        os.execute:revert()
-        utils.file_exists:revert()
     end)
 
-    it('test firmware_upgrade', function()
+    it('test firmware_upgrade without new metadata', function()
```

Comment:
@spiccinini these tests cover the metadata functionality partially, which is understandable given the effort needed to test it completely.
It could be nice to test that the metadata will be persisted. I think the best way to test it is with an integration test, but needs to mock fs and 'reboot' somehow.